packages/core/tsconfig.json

```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```

Comment:
Should the ESM version of the package use `moduleResolution: node16` or later here?

Reply:
What would be the best option for CJS?

Reply:
...and is esnext good for module and target?

Reply:
> tsconfig.json(6,15): error TS5110: Option 'module' must be set to 'Node16' when option 'moduleResolution' is set to 'Node16'.